traitlets/traitlets.py

```diff
@@ -392,8 +393,13 @@ class that does the following:
 
     name = None
     this_class = None
+    _parent = None
 
-    def class_init(self, cls, name):
+    @property
+    def absolute_name(self):
+        return list(self._lineage())[-1].name
+
+    def class_init(self, cls, name, parent=None):
```

Comment:
How do we handle custom TraitTypes that have defined `class_init(cls, name)` without breaking them?

Reply:
@minrk the best we can do is check the function signature I guess.